### PR TITLE
Feature/arm.v1

### PR DESCRIPTION
**Pull request by NWSchallock**
PTAL

### CODE REVIEW

File: src/Subsystems/Arm.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 #include "WPILib.h"
 #include "ctre/Phoenix.h"
 #include "AnalogInput.h"
+#include <Preferences.h>
```
Comment (duane701) [inline src/Subsystems/Arm.h]: Remove `#include <Preferences.h>`. 

`Preferences.h` is not needed in the header file. `Preferences.h` is needed in the source.

File: src/Subsystems/Arm.cpp [hunk 2]
```diff
@@ -2,6 +2,8 @@
 #include "../RobotMap.h"
```
Comment (duane701) [inline src/Subsystems/Arm.cpp]: `#include <Preferences.h>`

File: src/Commands/ArmMove.cpp [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+// Make this return true when this Command no longer needs to run execute()
+bool ArmMove::IsFinished() {
+	if (Arm::getInstance()->IsFwdLimitSwitchClosed() == true || Arm::getInstance()->IsRevLimitSwitchClosed() == true) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
```
Comment (duane701) [inline src/Commands/ArmMove.cpp]: Replace this entire function with

    return (Arm::getInstance()->IsFwdLimitSwitchClosed() || Arm::getInstance()->IsRevLimitSwitchClosed());

File: src/Commands/ArmPos.cpp [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+void ArmPos::Execute() {
+	Arm::getInstance()->SetArmPos(mPosition);
+}
+
+// Make this return true when this Command no longer needs to run execute()
+bool ArmPos::IsFinished() {
+	if (abs(Arm::getInstance()->GetPosError()) < 1000
+			|| Arm::getInstance()->IsFwdLimitSwitchClosed()
+			|| Arm::getInstance()->IsRevLimitSwitchClosed()) {
+		if (counter > 29) {
```
Comment (duane701) [inline src/Commands/ArmPos.cpp]: Le't's be careful with tabs and spaces.

Comment (duane701) [reply src/Commands/ArmPos.cpp]: What is `29`? 

I think this is determined later, so let's just leave this. But we need to name the value and say how we determined it's value.

Comment (duane701) [reply src/Commands/ArmPos.cpp]: `29` is how many times does the arm have to be in the right error before the arm is in the right position. 

We need a name for that. 😄

File: src/RobotMap.h [hunk 5]
```diff
@@ -26,6 +26,7 @@ class RobotMap {
 	  static const int kIDIntakeMotor = 5;
 	//..........Arm..........
 	  static const int kIDArmMotor = 10;
+	  static const int kIDArmPot = 3;
```
Comment (duane701) [inline src/RobotMap.h]: Let's distinguish between CAN IDs and other RoboRio ports. For example, `kIDArmPot` is an analog input on the RoboRio. Maybe use `kAIIDArmPot` or `kAnalogIDArmPot`?

**Review by duane701 — CHANGES_REQUESTED** [review]
FTP

**Comment by duane701** [comment]
PTAL @NWSchallock 

Look at the two commits I made.

**Comment by NWSchallock** [comment]
LGTM ♥

**Review by duane701 — APPROVED** [review]

**Pull request merged**